remember.go

```diff
@@ -0,0 +1,97 @@
+package remember
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+type Memory struct {
+	input []io.Reader
+	output io.ReadWriter
+}
+
+type Option func(*Memory) error
+
+func WithInput(args []string) Option {
+	return func(m *Memory) error {
+		var input = []io.Reader{}
+		for _, str := range args {
+			input = append(input, bytes.NewBufferString(str))
+		}
+		m.input = input
+		return nil
+	}
+}
+
+func WithOutput(output string) Option {
+	return func(m *Memory) error {
+		m.output = bytes.NewBufferString(output)
+		return nil
+	}
+}
+
+func NewMemory(opts ...Option) (Memory, error) {
+	_, err := os.Stat("store.txt")
+	if err != nil {
+		os.Create("store.txt")
+	}
+	defaultFile, err := os.OpenFile("store.txt", os.O_APPEND|os.O_RDWR, 0644)
```

Comment:
This error isn't checked, in an `if err != nil` statement, which can kick the error condition down the road. For example, if this `os.OpenFile()` call fails due to an uncaught `open store.txt: permission denied` error, the later call to `ioutil.ReadAll()` returns the less useful error `error opening data store: invalid argument`.